Sources/Common/Communication/Event.swift

```diff
@@ -24,71 +24,120 @@ public extension EventRepresentable {
     }
 }
 
+// MARK: - Event Types Registry
+
+/// Registry of all event types in the system.
+public enum EventTypesRegistry {
+    static func allEventTypes() -> [any EventRepresentable.Type] {
+        // Add all your event types here
+        [
+            ProfileIdentifiedEvent.self,
+            ScreenViewedEvent.self,
+            ResetEvent.self,
+            TrackMetricEvent.self,
+            TrackInAppMetricEvent.self,
+            RegisterDeviceTokenEvent.self,
+            DeleteDeviceTokenEvent.self,
+            NewSubscriptionEvent.self
+        ]
+    }
+}
+
+// MARK: - Event Structs
+
 public struct ProfileIdentifiedEvent: EventRepresentable {
+    public var storageId: String
     public var params: [String: String] = [:]
```

Comment:
public + mutable, not cool. quick path to threading issues

Reply:
The events are internal and usage of the events is done under dispatch queue only to make sure we are handling threading but i agree its still safe to make these immutable as we did in the protocol.

Reply:
even internal, if it smells it smells and going to cause pain sooner or later :)